purchases/src/androidTestIntegrationTest/kotlin/com/revenuecat/purchases/trustedentitlements/TrustedEntitlementsInformationalFailureIntegrationTest.kt

```diff
@@ -0,0 +1,81 @@
+package com.revenuecat.purchases.trustedentitlements
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.revenuecat.purchases.BasePurchasesIntegrationTest
+import com.revenuecat.purchases.CustomerInfo
+import com.revenuecat.purchases.EntitlementVerificationMode
+import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI
+import com.revenuecat.purchases.PurchaseParams
+import com.revenuecat.purchases.Purchases
+import com.revenuecat.purchases.VerificationResult
+import com.revenuecat.purchases.factories.StoreProductFactory
+import com.revenuecat.purchases.factories.StoreTransactionFactory
+import com.revenuecat.purchases.getCustomerInfoWith
+import com.revenuecat.purchases.helpers.mockQueryProductDetails
+import com.revenuecat.purchases.purchaseWith
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.fail
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
+@RunWith(AndroidJUnit4::class)
+class TrustedEntitlementsInformationalFailureIntegrationTest : BasePurchasesIntegrationTest() {
```

Comment:
Maybe `TrustedEntitlementsInformationalModeIntegrationTest`? Since you might want to test successes too?
Oh I see you have a separate one. I'm usually not against splitting test classes, easier to maintain. But I think it might be weird because it would be really useful to add tests that verify the behavior when the signature goes from correct to invalid and vice-versa. What do you think?

Reply:
Yeah I added one in the success file, but it's true it doesn't fully fit... I was torn because I didn't want to make a huge test class and it's easier to setup the condition in the `@Before` of the test, but well I think we don't have that many currently and we can indeed modify these things later in the integration test, so I will change it back to only one class.

Reply:
Done in https://github.com/RevenueCat/purchases-android/pull/1071/commits/dc6a1fa5f619b9eea16463671eef7d6afaa2af43